packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -189,8 +189,24 @@ export class RoutesMatcher {
 
 		const rewriteKey = 'x-middleware-rewrite';
 		const rewriteHeader = resp.headers.get(rewriteKey);
+
 		if (rewriteHeader) {
 			const newUrl = new URL(rewriteHeader, this.url);
+
+			const thisHost = this.url.hostname;
+			const rewriteHost = newUrl.hostname;
+			const rewriteIsExternal = thisHost !== rewriteHost;
+
+			if (rewriteIsExternal) {
+				const externalResp = await fetch(newUrl, this.reqCtx.request);
+				// The rewrite is external so we do not want to continue and we just
+				// set the final response body, status and headers here.
+				this.body = externalResp.body;
+				this.status = externalResp.status;
+				applyHeaders(this.headers.normal, externalResp.headers);
+				return;
```

Comment:
I don't think returning here is the right thing to do. In every other case, we are apply the middleware response's headers to headers.normal, but for this we are not. When it generates a response to return later on, it would apply the headers.normal to the final response, but since this is skipping updating headers.normal, it will never get other headers that were set in the middleware.

Reply:
good point, return removed, thanks 🙂